### PR TITLE
*: remove the previous joined pd

### LINKED ISSUE #7954: TestAPI/TestStores failed
### Which jobs are failing
TestAPI/TestStores

=== NAME  TestAPI/TestStores
    api_test.go:659: 
        	Error Trace:	/home/runner/work/pd/pd/tests/integrations/mcs/scheduling/api_test.go:659
        	            				/home/runner/work/pd/pd/tests/testutil.go:310
        	            				/home/runner/work/pd/pd/tests/integrations/mcs/scheduling/api_test.go:612
        	Error:      	Not equal: 
        	            	expected: 3
        	            	actual  : 2
        	Test:       	TestAPI/TestStores

### CI link
https://github.com/tikv/pd/actions/runs/8355487618/job/22870673004?pr=7951

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #7954.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/integrations/mcs/scheduling/api_test.go [hunk 1]
```diff
@@ -513,6 +513,15 @@ func (suite *apiTestSuite) checkFollowerForward(cluster *tests.TestCluster) {
 	defer cancel()
 	follower, err := cluster.JoinAPIServer(ctx)
 	re.NoError(err)
+	defer func() {
+		leader := cluster.GetLeaderServer()
+		cli := leader.GetEtcdClient()
+		_, err := cli.MemberRemove(ctx, follower.GetServer().GetMember().ID())
```
Comment (HuSharp) [inline tests/integrations/mcs/scheduling/api_test.go:519]: Should we remove the follower from `cluster.servers`, destroy the follower, and wait for Leader?

File: tests/integrations/mcs/scheduling/api_test.go [hunk 2]
```diff
@@ -513,6 +513,20 @@ func (suite *apiTestSuite) checkFollowerForward(cluster *tests.TestCluster) {
...
+		testutil.Eventually(re, func() bool {
+			_, err = cli.MemberRemove(context.Background(), follower.GetServer().GetMember().ID())
+			if err != nil {
+				return false
+			}
+			res, err := cli.MemberList(context.Background())
+			return err == nil && len(res.Members) == 1
+		})
+		cluster.DeleteServer(follower.GetConfig().Name)
+		follower.Destroy()
```
Comment (HuSharp) [inline tests/integrations/mcs/scheduling/api_test.go:528]: Since this is an `api suite`, maybe we'd better add the `waitLeader` function as well?

Comment (rleungx) [reply tests/integrations/mcs/scheduling/api_test.go:528]: We only remove the follower which won't affect the leader.

Comment (HuSharp) [reply tests/integrations/mcs/scheduling/api_test.go:528]: oh, you are right

**Review by HuSharp — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Comment by HuSharp** [comment]
ci failed because meeting panic :(
https://github.com/tikv/pd/actions/runs/10570839546/job/29285961014?pr=8562

```bash
2024-08-27T02:54:35.0538010Z [FAIL]  mcs/scheduling TestAPI
2024-08-27T02:54:35.2144314Z === RUN   TestAPI/TestFollowerForward
2024-08-27T02:54:35.2144959Z === NAME  TestAPI
2024-08-27T02:54:35.2145730Z     testutil.go:337: start test TestFollowerForward in pd mode
...
2024-08-27T02:54:35.8353903Z [2024/08/27 02:54:34.947 +00:00] [PANIC] [cluster.go:859] ["failed to find local ID in cluster members"] [cluster-id=b2fe976be7a757da] [local-member-id=6c281856cbb0ddbe]
2024-08-27T02:54:35.8354083Z panic: failed to find local ID in cluster members
```

File: tests/integrations/mcs/scheduling/api_test.go [hunk 3]
```diff
@@ -513,6 +513,20 @@ func (suite *apiTestSuite) checkFollowerForward(cluster *tests.TestCluster) {
 	defer cancel()
 	follower, err := cluster.JoinAPIServer(ctx)
 	re.NoError(err)
+	defer func() {
+		leader := cluster.GetLeaderServer()
+		cli := leader.GetEtcdClient()
+		testutil.Eventually(re, func() bool {
```
Comment (lhy1024) [inline tests/integrations/mcs/scheduling/api_test.go:519]:
```suggestion
_, err = cli.MemberRemove(context.Background(), follower.GetServer().GetMember().ID())
			re.NoError(err)
		testutil.Eventually(re, func() bool {
			res, err := cli.MemberList(context.Background())
			return err == nil && len(res.Members) == 1
		})
```

Comment (rleungx) [reply tests/integrations/mcs/scheduling/api_test.go:519]: I have split it into two `Eventually`.

**Pull request merged**